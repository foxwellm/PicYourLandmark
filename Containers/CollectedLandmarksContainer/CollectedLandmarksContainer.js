import React, {Component} from 'react';
import { StyleSheet, Text, View, ImageBackground, ScrollView } from 'react-native';
import { Icon } from 'react-native-elements';
import Landmark from '../Landmark/Landmark';

export class CollectedLandmarksContainer extends Component {
    render() {
        return (
            <View style={styles.container}>
                <View style={{flex: 1}}>
                <ScrollView style={styles.scrollView}>
                    <ImageBackground source={require('../../assets/greatwallofchina.jpg')} style={styles.imageBackground}>
                    <View style={styles.overlay} />
                    <Text style={styles.bannerText}>COLLECTED LANDMARKS</Text>
                    <Icon color="white" name="university" type="font-awesome" size={40} top={-110}/>
                    </ImageBackground>
                    {
                        this.props.visitedLocations.map((location) => {
                            return (<Landmark key={location.landmark_id} landmark={location}/>)
                        })
                    }
                </ScrollView>
            </View>
        </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1
    },
    overlay: {
        position: 'absolute',
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
        backgroundColor: 'black',
        opacity: 0.4,
        height: '54%',
    },
    imageBackground: {
        width: '100%', 
        height: '54%', 
        display: 'flex', 
        flexDirection: 'column', 
        justifyContent: 'center', 
        alignItems: 'center'
    },
    bannerText: {
        color: 'white',
        fontSize: 18,
        fontWeight: '600',
        top: -100,
        padding: 10
    },
    scrollView: {
        height: '300%'
    }
});

export default CollectedLandmarksContainer;