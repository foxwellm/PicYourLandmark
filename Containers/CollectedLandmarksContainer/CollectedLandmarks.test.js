import React from 'react';
import { CollectedLandmarksContainer } from './CollectedLandmarksContainer';
import renderer from 'react-test-renderer';

describe('CollectedLandmarksContainer', () => {
  let mockVisitedLocations;
  it('renders correctly', () => {
    mockVisitedLocations = [
      {
        name: "Great Lawn Park",
        description: "Beautiful Park",
        lat: 39.72386,
        lon: -104.88715,
        landmark_id: 6,
        photo_url: "www.myimage.com/2"
    },
    {
        name: "Buckley Annex",
        description: "Beautiful Park",
        lat: 39.7159,
        lon: -104.90379,
        landmark_id: 3,
        photo_url: "www.myimage.com"
    }
  ]
    const tree = renderer.create(<CollectedLandmarksContainer visitedLocations={mockVisitedLocations}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
