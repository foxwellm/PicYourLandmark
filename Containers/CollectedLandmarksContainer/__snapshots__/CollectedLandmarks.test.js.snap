// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectedLandmarksContainer renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexGrow": 1,
        },
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />
</View>
`;
